homeassistant/components/microbees/switch.py

```diff
@@ -0,0 +1,81 @@
+"""Switch integration microBees."""
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import BEES, CONNECTOR, COORDINATOR, DOMAIN
+from .coordinator import MicroBeesEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Config entry."""
+    microbees = hass.data[DOMAIN][CONNECTOR]
+    bees = hass.data[DOMAIN][BEES]
+    coordinator = hass.data[DOMAIN][COORDINATOR]
+    switches = []
+    for bee in bees:
+        match bee.productID:
+            case 25 | 26 | 27 | 35 | 38 | 46 | 63 | 64 | 65 | 86:
+                for switch in bee.actuators:
+                    switches.append(MBSwitch(switch, bee, microbees, coordinator))
+
+    async_add_entities(switches)
+
+
+class MBSwitch(MicroBeesEntity, SwitchEntity):
+    """Representation of a microBees switch."""
+
+    def __init__(self, act, bee, microbees, coordinator) -> None:
+        """Initialize the microBees switch."""
+        self.act = act
+        self.bee = bee
+        self.microbees = microbees
+        self._state = self.act.value
+        self._attr_available = self.bee.active
+        self._coordinator = coordinator
+        self._attr_unique_id = self.act.id
+        self._attr_name = self.act.name + " (" + self.bee.name + ")"
```

Comment:
Can you maybe explain what an actuator is and what a bee is? (in this context?)

Reply:
You need to fix your naming then.

New integrations are required to use `_attr_has_entity_name = True` 

Then create a `_attr_device_info` with the device identifier (the bee in this case) and the bee name as name and then the name of this switch should only have to be the actuator name

Reply:
ok, solved, also this, thank you